sdk/lib/dmilib/dmilib.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * PROJECT:     ReactOS DMI/SMBIOS Library
+ * LICENSE:     GPL - See COPYING in the top level directory
+ * FILE:        dmilib.c
+ * PURPOSE:     SMBIOS table parsing functions
+ * PROGRAMMERS: Timo Kreuzer (timo.kreuzer@reactos.org)
+ * REFERENCES:  - http://www.dmtf.org/sites/default/files/standards/documents/DSP0134_3.0.0.pdf
+ *              -
+ */
+
+#include "precomp.h"
+
+PCHAR Strings[ID_STRINGS_MAX] = { 0 };
```

Comment:
Because you moved this to function local scope to global scope,
it is now inherently un-threadsafe.

Reply:
Any suggestion how to fix it, some example code?

By the way, these functions are not expected to be called from different threads simultaneously.

Reply:
In this specific case, I would make the user in charge of providing the `Strings` array, cutting out the `GetParsedDmiString` function entirely.

Reply:
Ok, done.